app/forms/claims/mentor_form.rb

```diff
@@ -19,15 +27,30 @@ def as_form_params
   end
 
   def mentor
-    @mentor ||= MentorBuilder.call(trn:, first_name:, last_name:).becomes(Claims::Mentor)
+    @mentor ||= Claims::MentorBuilder.call(
+      trn:,
+      first_name:,
+      last_name:,
+      date_of_birth:,
+    ).becomes(Claims::Mentor)
```

Comment:
A `Claims::MentorBuilder` should be able to handle the type-casting within the service, as it's already namespaced.